packages/api/src/types.ts

```diff
@@ -84,18 +91,22 @@ export type DecoratedRpc<ApiType, AllSections> = {
   [Section in keyof AllSections]: DecoratedRpcSection<ApiType, AllSections[Section]>
 }
 
-interface StorageEntryBase<C, H, U> {
-  at: (hash: Hash | Uint8Array | string, arg1?: CodecArg, arg2?: CodecArg) => C;
-  creator: StorageEntry;
-  hash: (arg1?: CodecArg, arg2?: CodecArg) => H;
-  key: (arg1?: CodecArg, arg2?: CodecArg) => string;
-  size: (arg1?: CodecArg, arg2?: CodecArg) => U;
+interface StorageEntryObservableMulti {
+  <T extends Codec>(args: (CodecArg[] | CodecArg)[]): Observable<T[]>;
+}
+
+interface StorageEntryPromiseMulti {
+  <T extends Codec>(args: (CodecArg[] | CodecArg)[]): Promise<T[]>;
+  <T extends Codec>(args: (CodecArg[] | CodecArg)[], callback: Callback<T[]>): UnsubscribePromise;
 }
 
-export interface StorageEntryObservable extends StorageEntryBase<Observable<Codec>, Observable<Hash>, Observable<u64>> {
-  (arg1?: CodecArg, arg2?: CodecArg): Observable<Codec>;
-  <T extends Codec>(arg1?: CodecArg, arg2?: CodecArg): Observable<T>;
-  multi: <T extends Codec>(args: (CodecArg[] | CodecArg)[]) => Observable<T[]>;
+export interface StorageEntryBase<ApiType, F extends AnyFunction> {
+  at: (hash: Hash | Uint8Array | string, ...args: Parameters<F>) => PromiseOrObs<ApiType, ObsInnerType<ReturnType<F>>>;
+  creator: StorageEntry;
+  hash: (...args: Parameters<F>) => PromiseOrObs<ApiType, Hash>;
+  key: (...args: Parameters<F>) => string;
+  size: (...args: Parameters<F>) => PromiseOrObs<ApiType, u64>;
+  multi: ApiType extends 'rxjs' ? StorageEntryObservableMulti : StorageEntryPromiseMulti;
```

Comment:
Didn't see a multi test. However this does seem to be a `Codec[]` result -

`const balances = await api.query.balances.freeBalance.multi([keyring.alice.address, keyring.bob.address, keyring.eve.address]);`

Reply:
I added a .multi test. It should be a `u128[]`, but I cannot type that correctly, so it'll stay a `Codec[]` for now.

This works:
```
const balances = await api.query.balances.freeBalance.multi<Balance>([keyring.alice.address, keyring.bob.address, keyring.eve.address]);
```

Reply:
That is fine. There must be a way of doing this on the `.multi` though, `queryMulti` probably not. Not a crisis since this is 100x better already, but something to keep in mind.